aws_cloudwatch_event_target panics if sqs_target block is empty

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform AWS Provider Version

```
 [fg-386] tf_sqs_issue_repro > terraform version
Terraform v0.15.5
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.46.0

Your version of Terraform is out of date! The latest version
is 1.0.0. You can update by downloading from https://www.terraform.io/downloads.html
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_cloudwatch_event_target 

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
terraform {
  required_version = ">= 0.14"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 3.25"
    }
  }
}

locals {
  message_group_id = null
}

terraform {
  backend "local" {}
}

provider "aws" {
  alias  = "va"
  region = "us-east-1"
}

resource "aws_sqs_queue" "test_queue" {
  provider = aws.va
  name     = "terraform_test_queue"
}

resource "aws_cloudwatch_event_rule" "test_rule" {
  provider            = aws.va
  name                = "test_rule"
  description         = "Send a message at a set interval to kick off a job."
  schedule_expression = "cron(0 12 * * ? *)"
}

resource "aws_cloudwatch_event_target" "test_rule_target" {
  provider = aws.va
  rule     = aws_cloudwatch_event_rule.test_rule.name
  arn      = aws_sqs_queue.test_queue.arn

  input = jsonencode({
    foo = "bar"
  })

  sqs_target {
    message_group_id = local.message_group_id
  }
}
```

### Debug Output

https://gist.github.com/packrat386/b88da9175a551d4ab5968c8c87387887

### Panic Output

```
│ Error: Plugin did not respond
│
│   with aws_cloudwatch_event_target.test_rule_target,
│   on main.tf line 36, in resource "aws_cloudwatch_event_target" "test_rule_target":
│   36: resource "aws_cloudwatch_event_target" "test_rule_target" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-aws_v3.46.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 195 [running]:
github.com/terraform-providers/terraform-provider-aws/aws.expandAwsCloudWatchEventTargetSqsParameters(0xc001692900, 0x1, 0x1, 0x593e8c0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_cloudwatch_event_target.go:673 +0x18b
github.com/terraform-providers/terraform-provider-aws/aws.buildPutTargetInputStruct(0xc0013fd500, 0x7128d4f)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_cloudwatch_event_target.go:518 +0x507
github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsCloudWatchEventTargetCreate(0xc0013fd500, 0x632bc80, 0xc0012ac000, 0x0, 0xffffffffffffffff)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_cloudwatch_event_target.go:334 +0x18f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0007a3420, 0x7c74908, 0xc00150d5c0, 0xc0013fd500, 0x632bc80, 0xc0012ac000, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:318 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0007a3420, 0x7c74908, 0xc00150d5c0, 0xc000b1fc70, 0xc001027660, 0x632bc80, 0xc0012ac000, 0x0, 0x0, 0x0, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000306780, 0x7c74908, 0xc00150d5c0, 0xc00153e7d0, 0xc00150d5c0, 0x6f07b20, 0xc001563600)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0016bcee0, 0x7c749b0, 0xc00150d5c0, 0xc000b1fb20, 0xc0016bcee0, 0xc0015636e0, 0xc00105fba0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x6f07b20, 0xc0016bcee0, 0x7c749b0, 0xc0015636e0, 0xc0015469c0, 0x0, 0x7c749b0, 0xc0015636e0, 0xc001dcdb00, 0x2c6)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005b56c0, 0x7c963b8, 0xc001b0a180, 0xc001393a00, 0xc0011c3050, 0xb402a60, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0005b56c0, 0x7c963b8, 0xc001b0a180, 0xc001393a00, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000a68cd0, 0xc0005b56c0, 0x7c963b8, 0xc001b0a180, 0xc001393a00)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-aws_v3.46.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

This should not crash and add a cloudwatch event target to an SQS queue with no message_group_id.

### Actual Behavior

It crashed

### Steps to Reproduce

1. `terraform apply`

https://github.com/packrat386/tf_sqs_issue_repro

### Important Factoids

N/A

------

I'm aware that this is a rather silly bug, as one should avoid adding the block if one does not have a message_group_id, but the provider still ought to handle it gracefully.

--- a/.changelog/19946.txt
+++ b/.changelog/19946.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_cloudwatch_event_target: Don't crash if `sqs_target` configuration block is empty.
+```

--- a/aws/resource_aws_cloudwatch_event_target.go
+++ b/aws/resource_aws_cloudwatch_event_target.go
@@ -494,11 +494,11 @@ func buildPutTargetInputStruct(d *schema.ResourceData) *events.PutTargetsInput {
 		e.RoleArn = aws.String(v.(string))
 	}
 
-	if v, ok := d.GetOk("run_command_targets"); ok {
+	if v, ok := d.GetOk("run_command_targets"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 		e.RunCommandParameters = expandAwsCloudWatchEventTargetRunParameters(v.([]interface{}))
 	}
 
-	if v, ok := d.GetOk("ecs_target"); ok {
+	if v, ok := d.GetOk("ecs_target"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 		e.EcsParameters = expandAwsCloudWatchEventTargetEcsParameters(v.([]interface{}))
 	}
 
@@ -506,27 +506,27 @@ func buildPutTargetInputStruct(d *schema.ResourceData) *events.PutTargetsInput {
 		e.HttpParameters = expandAwsCloudWatchEventTargetHttpParameters(v.([]interface{})[0].(map[string]interface{}))
 	}
 
-	if v, ok := d.GetOk("batch_target"); ok {
+	if v, ok := d.GetOk("batch_target"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 		e.BatchParameters = expandAwsCloudWatchEventTargetBatchParameters(v.([]interface{}))
 	}
 
-	if v, ok := d.GetOk("kinesis_target"); ok {
+	if v, ok := d.GetOk("kinesis_target"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 		e.KinesisParameters = expandAwsCloudWatchEventTargetKinesisParameters(v.([]interface{}))
 	}
 
-	if v, ok := d.GetOk("sqs_target"); ok {
+	if v, ok := d.GetOk("sqs_target"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 		e.SqsParameters = expandAwsCloudWatchEventTargetSqsParameters(v.([]interface{}))
 	}
 
-	if v, ok := d.GetOk("input_transformer"); ok {
+	if v, ok := d.GetOk("input_transformer"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 		e.InputTransformer = expandAwsCloudWatchEventTransformerParameters(v.([]interface{}))
 	}
 
-	if v, ok := d.GetOk("retry_policy"); ok {
+	if v, ok := d.GetOk("retry_policy"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 		e.RetryPolicy = expandAwsCloudWatchEventRetryPolicyParameters(v.([]interface{}))
 	}
 
-	if v, ok := d.GetOk("dead_letter_config"); ok {
+	if v, ok := d.GetOk("dead_letter_config"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 		e.DeadLetterConfig = expandAwsCloudWatchEventDeadLetterConfigParameters(v.([]interface{}))
 	}
 

